src/Payment/PaymentTests/DALTests/PaymentRepositoryTests.cs

```diff
@@ -0,0 +1,129 @@
+﻿using DAL;
+using DAL.Events;
+using DAL.Payments;
+using DAL.Payments.Repository;
+using Microsoft.Extensions.DependencyInjection;
+using TestsCore;
+
+namespace PaymentTests.DALTests
+{
+	public class PaymentRepositoryTests
+	{
+		[Theory]
+		[InlineData(GetValuesSuites.Empty)]
+		[InlineData(GetValuesSuites.OneValue)]
+		[InlineData(GetValuesSuites.ThreeValues)]
+		[InlineData(GetValuesSuites.ManyValues)]
+		public async Task FailPayment_ExistingValues_SeatsAvailable(GetValuesSuites suite)
+		{
+			await ProcessPayment_ExistingValues_Status(suite,
+				complete: false,
+				PaymentStatus.Failed.ToString().ToLowerInvariant(),
+				SeatStatus.Available.ToString().ToLowerInvariant());
+		}
+
+		[Theory]
+		[InlineData(GetValuesSuites.Empty)]
```

Comment:
Am I understanding correctly, that we are trying to complete payment for no seats? Should payment be created at all in this case?

Reply:
No, it shouldn't, I'll delete this case